pkg/batch/jobs/calculator/experiment_calculate.go

```diff
@@ -154,8 +156,7 @@ func (e *experimentCalculate) calculateExperimentWithLock(ctx context.Context,
 		)
 		return nil
 	}
-
-	defer func() {
+	if calcErr := e.calculateExperiment(ctx, env, experiment); calcErr != nil {
```

Comment:
```suggestion
	if calcErr := e.calculateExperiment(ctx, env, experiment); calcErr != nil {
	    // To prevent calculating the same experiment multiple times in a short time,
	    // we set the TTL for the lock key and only unlock it when an error occurs.
```